src/hotspot/cpu/x86/x86.ad

```diff
@@ -8697,12 +8688,11 @@ instruct vpopcountI_avx_reg(vec dst, vec src, vec xtmp1, vec xtmp2, rRegP rtmp)
 %}
 
 instruct vpopcountL_avx_reg(vec dst, vec src, vec xtmp1, vec xtmp2, vec xtmp3, rRegP rtmp) %{
-  predicate(!VM_Version::supports_avx512_vpopcntdq());
+  predicate(!UsePopCountInstruction || !VM_Version::supports_avx512_vpopcntdq());
```

Comment:
Please remove the reference to UsePopCountInstruction here.

Reply:
There is still a matcher path which can be chosen even if user turns off UsePopCountInstruction,  relating this flag to an operation may not be correct.  But I agree that to avoid any confusion we can remove its usage from pattern for the time being.